frida_mode/src/instrument/instrument_arm64.c

```diff
@@ -1,101 +1,221 @@
+#include <stddef.h>
+
 #include "frida-gumjs.h"
 
 #include "config.h"
 
 #include "instrument.h"
+#include "ranges.h"
+#include "stalker.h"
+#include "util.h"
+
+#define G_MININT33 ((gssize)0xffffffff00000000)
+#define G_MAXINT33 ((gssize)0x00000000ffffffff)
+
+#define PAGE_MASK (~(GUM_ADDRESS(0xfff)))
+#define PAGE_ALIGNED(x) ((GUM_ADDRESS(x) & PAGE_MASK) == GUM_ADDRESS(x))
 
 #if defined(__aarch64__)
 
-static GumAddress current_log_impl = GUM_ADDRESS(0);
+__attribute__((aligned(0x1000))) static guint8 area_ptr_dummy[MAP_SIZE];
+
+  #pragma pack(push, 1)
+typedef struct {
+
+  // cur_location = (block_address >> 4) ^ (block_address << 8);
+  // shared_mem[cur_location ^ prev_location]++;
+  // prev_location = cur_location >> 1;
```

Comment:
@WorksButNotTested this looks to revert the `ROR` logic (`// previous_pc = current_pc ROR 1;`) from 94999782f1a3742e3e755a66f5d76e84573ae6ef that is described in the [MapDensity.md](https://github.com/WorksButNotTested/AFLplusplus/blob/376d1736a8f9de28e76c739c448ada2fb4fcb5a7/frida_mode/MapDensity.md), was this intended?

Reply:
It's a long time since I made this change, but I think the calculation remains the same. Please see the comment below.